### PR TITLE
Add settle command

### LINKED ISSUE #127: Add settle command

### PR DESCRIPTION
**Pull request by Soumyadip-cmd**
Closes #127 

Implement SettleCommand and SettleCommandParser to 
update student payment records

The code only supports commands to track the amount owed 
and the total paid, without a way to settle specific payment 
amounts for students. This limits the functionality for users to 
manage student payments effectively, especially when partial 
settlements are required.

Let's
1. Add settle command and settle command parser
2.  Add functionality to update both paidAmount 
        and owedAmount
3. Add checks for amount paid to ensure it is a positive 
        number and is not more than owedAmount
5. Add tests to verify functionality of settle command 
        and settle command parser

image

image

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/SettleCommand.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+ */
+public class SettleCommand extends Command {
+
+    public static final String COMMAND_WORD = "settle";
+    public static final String COMMAND_WORD_RANDOM_CASE = "sEttLe";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Updates owed amount and paid amount of the student "
+            + "identified by the index number used in the displayed Student list.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_AMOUNT + "AMOUNT] ";
```
Comment (TY1Fan) [inline src/main/java/seedu/address/logic/commands/SettleCommand.java:30]: I though square brackets mean optional parameters?

Comment (Soumyadip-cmd) [reply src/main/java/seedu/address/logic/commands/SettleCommand.java]: You are right. Sorry about that. I have fixed the error.

**Review by teamotitan — APPROVED** [review]
It looks good to me

**Review by TY1Fan — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/logic/parser/SettleCommandParserTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    @Test
+    public void parse_multipleRepeatedFields_failure() {
+        String userInput = INDEX_FIRST_STUDENT.getOneBased() + VALID_AMOUNT_DESC + INVALID_AMOUNT_DESC;
+
+        assertParseFailure(parser, userInput, Messages.getErrorMessageForDuplicatePrefixes(PREFIX_AMOUNT));
+    }
+
+    // Helper methods for assertions
+
+    private void assertParseFailure(SettleCommandParser parser, String userInput, String expectedMessage) {
```
Comment (Feng1231) [inline src/test/java/seedu/address/logic/parser/SettleCommandParserTest.java:76]: Hmm, isn't this already defined in `CommandParserTestUtil.java`?

File: src/test/java/seedu/address/logic/parser/SettleCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        ParseException thrown = assertThrows(ParseException.class, () -> parser.parse(userInput));
+        assertEquals(expectedMessage, thrown.getMessage());
+    }
+
+    private void assertParseSuccess(SettleCommandParser parser, String userInput, SettleCommand expectedCommand) {
```
Comment (Feng1231) [inline src/test/java/seedu/address/logic/parser/SettleCommandParserTest.java:81]: likewise

**Review by Feng1231 — COMMENTED** [review]
Nice, very comprehensive! Having some logging would be nice.
Also, the `assertParseFailure` and Success methods are already defined in `CommandParserTestUtil`, I believe there's no need to redefine it.

File: src/test/java/seedu/address/logic/parser/SettleCommandParserTest.java (hunk 2 above)
Comment (Soumyadip-cmd) [reply src/test/java/seedu/address/logic/parser/SettleCommandParserTest.java]: Thank you for pointing that out. I have made changes accordingly.

**Review by Feng1231 — APPROVED** [review]
Nice changes! LGTM

**Pull request merged**